### ISSUE TITLE
Drawing attention to PL.Editor [Idea]

### ISSUE THREAD

**Issue by rexagod** [publiclab/PublicLab.Editor#193]
**Greetings!**

I have been knee-deep in PublicLab's Editor repository(love @jywarren's work there) for a while now. For all it is currently, I think it is a critical part of PL and showcases a great potential and with ample amout of carefully curated issues, this editor could be improved in a much more versatile and convenient manner.

Currently I'm thinking about testing out two new enhancements for this editor namely, **a LaTeX addition** along with Markdown and Rich Text (publishing PDFs along with other relevent text content), and, stay with me on this, _**a shared behaviour**_ that allows users to share a generated code so that others can collaborate with them, tell me that wouldn't be awesome!

These will take time, as all good things do, a little more than it's supposed to in fact, as I'm currently busy on a Native Electron implementation(focusing on Linux desktops for now) for the editor while keeping the code and behavioural changes to a minimum as @jywarren was so kind to point that out to me.

The only thing that bugs me though, is the ratio of the amout of engagement this repository is getting as of now and how much it deserves, tbh. I do believe that with the forementioned implementations, and a little "push" from the mentors, this can be resolved. But other than that, I will continue to contribute to this repository, and check for myself which of the other JavaScript repositories interests me the most.

Sorry for this being so lengthly, but this was something that was on my mind for a while now.

Cheers PL, let's do this! :tada:

**Comment by SidharthBansal** [comment]
I really loved your idea. Public Lab Editor is a major project for plots2. We need to create multiple supportive tasks in this repository like we did in the plots2. Here is a link:https://github.com/publiclab/plots2/wiki/Supportive-Tasks
We need to plan so that this repository can become more famous. We included this in GSoC every year but no contributor was interested in this repository. It will be great if we can have this in RGSoC, GSoC this year also. In future we can include them in hackoctober too. Whar do you suggest @jywarren?

**Comment by SidharthBansal** [comment]
>>a LaTeX addition along with Markdown and Rich Text (publishing PDFs along with other relevent text content)

If Jeff agrees then we can form up a Planning issue for this and break it down into smaller issues.

**Comment by jywarren** [comment]
Yes i think this is really nice. I do think perhaps we could build out some initial community base by focusing on resolving some long-standing bugs and issues like #44, #59, #67... we could attack these through writing tests for them!

Also in the more distant future could be some convergence with the new Gutenberg editor as in #149. 

Thanks for your energy and very excited, @rexagod !!!

**Comment by jywarren** [comment]
How would the PDF bit work? Very interesting!!!

**Comment by jywarren** [comment]
I think this is an interesting idea but the implementation we have currently is based on 2 libs which convert back and forth from Markdown to HTML; `megamark` and `domador` -- to incorporate LaTeX into this we would have to be able to implement 4 more converters - one in each direction between Markdown and LaTeX, and two more between HTML and LaTeX. So this would be a pretty gigantic project, and as we've seen in some bugs we've been chasing, the Markdown/HTML conversion is not 100% solid in the first place. I think we can consider this project, but we'd have to:

1. search for existing conversion utilities to fulfill the 4 conversions above
2. run the same set of conversion tests that `megamark` and `domador` implement
3. integrate them into `woofmark` ideally so that it happens upstream of our editor

I'm wondering if, long-term, we need to explore more widely supported WYSIWYG libraries such as Gutenberg - #149 - so we may want to think hard about this. 

For example, `woofmark` and `horsey` integration didn't work that well and is not widely supported, which led us to invest time in the `at` library for autocompletion -- still not complete in #44 -- so we should keep in mind the community size and sustainability of upstream libraries. 

Actually if you're interested we'd love help with #44 -- you've been doing fantastic work in this library and we really appreciate it!

**Comment by jywarren** [comment]
Also - implementing a `Save as PDF` option sounds really nice, too! Although perhaps we should consider implementing it downstream, as an output option for `plots2` -- just to brainstorm, could we implement a client-side library which accepts Markdown input and produces a PDF, and integrate it on the output of a given note or wiki page from `plots2`?

**Comment by jywarren** [comment]
Great brainstorming here!

**Comment by rexagod** [comment]
@jywarren In response to your comment [here](https://github.com/publiclab/PublicLab.Editor/issues/193), I think,

- Gutenberg features can be robustly reproduced in React.
- At.js (no longer maintained), suggest using [zurb tribute's implementation](https://github.com/BolsteDev/react-tribute)(again in React).
- #59 is currently under progress and I am looking into it at #208 
- #67 is resolved and up for you to review at #215

**Comment by geekychaser** [comment]
Created a first timer issue #219 to increase involvement of more contributors

**Comment by niravasher** [comment]
Hello! Can I work on adding the 'Save as PDF' function for PublicLab.Editor? First I will enter all the details on the editor and on clicking save as pdf button, a pdf file will be downloaded which will have the copy of examples/index.html page completely filled?
